### ISSUE TITLE
Why should the batch size be set to 1?

### ISSUE THREAD

**Issue by chenhang98** [Yuanshi9815/OminiControl#127]
Thank you for your excellent work! 
I noticed that in both the paper and the configuration files, the batch size is set to 1, while the gradient accumulation steps are set to 8 or 4. This should be equivalent to batch_size=8 or 4, but with slower training speed - is this understanding correct?
As mentioned in this section of the [documentation](https://github.com/Yuanshi9815/OminiControl/tree/main/train#batch-size), this configuration helps stabilize training. I am curious about the rationale behind this design. Could you share any insights? Or is this purely an empirical configuration that yields better results?

**Comment by Yuanshi9815** [comment]
Hi @chenhang98 ,

Thank you for your question!

In most training scenarios, using batch sizes larger than 1 is feasible and equivalent to smaller batch sizes with gradient accumulation. However, in specific situations within our implementation, batch sizes greater than 1 might cause issues, particularly when dealing with:
1. Images with varying resolutions in a single batch.
2. Condition sequences of different lengths (when using token integration).
3. Different position encoding methods within a single batch.
This is because our current implementation and diffusers' code do not efficiently handle these special cases simultaneously.

Given the FLUX model's intensive resource usage, a batch size of 1 does not significantly slow down training. Therefore, we recommend using gradient accumulation for stability and simplicity. However, if your training does not involve these special cases, you can safely experiment with larger batch sizes. Additionally, if your GPU has sufficient memory, you can disable gradient checkpointing to speed up training.

;D

**Comment by chenhang98** [comment]
I see. Thank you for the detailed explanation.

**Comment by fikry102** [comment]
```
                c_ids[:, 1] += p_delta[0][0]
                c_ids[:, 2] += p_delta[0][1]
                if len(p_delta) > 1:
                    print("Warning: only the first position delta is used.")
```


I want to use batch_size=2, but some warning happens:


omini/train_flux/trainer.py Line 198-201

why choose p_delta[0] rather than  p_delta[:, 0]
c_ids[:, 1] += p_delta[0][0]
c_ids[:, 2] += p_delta[0][1]
